### PR TITLE
Simple purescript API call OBSOLETE

### PR DESCRIPTION
**Pull request by jkarni**

### CODE REVIEW

File: thentos-core/purescript/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/bower_components/
```
Comment (fhartwig) [inline thentos-core/purescript/.gitignore]: I think this was created by pulp (?), but we may want to stick to a single .gitignore for the whole project?

File: thentos-core/purescript/bower.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "foo",
```
Comment (fhartwig) [inline thentos-core/purescript/bower.json]: We can probably come up with a more descriptive package name.

File: thentos-core/purescript/.gitignore [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+/node_modules/
+/output/
+/.psci*
+/src/.webpack.js
```
Comment (fisx) [inline thentos-core/purescript/.gitignore]: does this still exist?  that would probably mean that somebody around here is already using webpack...

File: thentos-core/src/Thentos/Backend/Api/Simple.hs [hunk 4]
```diff
@@ -31,22 +32,26 @@ import Thentos.Backend.Api.Auth
...
 
-type Api = ThentosAssertHeaders :> ThentosAuth :> ThentosBasic
+type Api =
+         (ThentosAssertHeaders :> ThentosAuth :> ThentosBasic)
+    :<|> ("js" :> ThentosFrontend)
 
-api :: ActionState -> Server Api
-api actionState mTok = enter (enterAction actionState baseActionErrorToServantErr mTok) thentosBasic
+api :: ActionState -> FilePath -> Server Api
+api actionState basePath = (\mTok -> enter (enterAction actionState baseActionErrorToServantErr mTok) thentosBasic) :<|> thentosFrontend basePath
```
Comment (fisx) [inline thentos-core/src/Thentos/Backend/Api/Simple.hs]: inline `serveDirectory` here?

**Comment by fhartwig** [comment]
@jkarni Looks good, modulo minor comments.

File: thentos-core/src/Thentos/Backend/Api/Docs/Common.hs [hunk 5]
```diff
@@ -91,6 +91,11 @@ instance ToSample UserFormData UserFormData where
                               <*> toSample (Proxy :: Proxy UserPass)
                               <*> toSample (Proxy :: Proxy UserEmail)
 
+-- FIXME: long request bodys should be pretty-printed
```
Comment (jkarni) [inline thentos-core/src/Thentos/Backend/Api/Docs/Common.hs]: I don't really understand this comment (also there's a typo).

File: thentos-core/purescript/test/Main.purs [hunk 6]
```diff
@@ -0,0 +1,6 @@
+module Test.Main where
+
+import Control.Monad.Eff.Console
+
+main = do
+  log "You should add some tests."
```
Comment (jkarni) [inline thentos-core/purescript/test/Main.purs]: I'll remove this? I'm all in favour of adding tests, of course, but right now the only thing we'd be testing is the backend.

File: thentos-core/src/Thentos/Backend/Api/Simple.hs [hunk 7]
```diff
@@ -69,6 +74,7 @@ thentosBasic =
 
 type ThentosUser =
        ReqBody '[JSON] UserFormData :> Post '[JSON] UserId
+  :<|> "login" :> ReqBody '[JSON] LoginFormData :> Post '[JSON] ThentosSessionToken
```
Comment (jkarni) [inline thentos-core/src/Thentos/Backend/Api/Simple.hs]: Add test

Comment (jkarni) [inline thentos-core/src/Thentos/Backend/Api/Simple.hs]: Don't use top-level string for response.

File: thentos-core/purescript/src/Main.purs [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+instance showThentosError :: Show ThentosError where
+    show = gShow
+
+data LoginRequestBody = LoginRequestBody
+    { user :: Username
+    , pass :: Password
+    }
+
+instance loginRequestBodyToJSON :: ToJSON LoginRequestBody where
+  toJSON (LoginRequestBody { user: n, pass: p }) = object ["ldName" .= n, "ldPassword" .= p]
```
Comment (jkarni) [inline thentos-core/purescript/src/Main.purs]: @fisx why `ld`? Seems like an artefact of the backend not wanting record name conflicts?

**Comment by jkarni** [comment]
I'm in favour of waiting until we've discussed things tomorrow before merging this.

File: thentos-core/purescript/src/Main.purs (hunk 8 above)
Comment (fisx) [inline thentos-core/purescript/src/Main.purs]: you are right.  both with your guess and with your opinion that that looks a little sloppy.  if you want me to fix the `FromJSON` instance in the backend or if you have another idea let me know.

**Comment by fisx** [comment]
I'm in favour of merging.  It works, it is simple, and it is a very small part of this pr.  and it will change again over time anyway.

**Comment by jkarni** [comment]
I also don't like how basically we're developing two endpoints for the same resource/behaviour (logging in) just one outputting HTML and the other JSON, and taking fairly different code paths (even different frameworks!). It'd be nice to think of a less ad-hoc direction to take.

**Comment by fisx** [comment]
@jkarni: re your last comment: see #229, #228.

**Comment by ChristianSi** [comment]
What's the status of this PR? It has been open for a long time.

**Comment by fisx** [comment]
I left it open because I want to implement US#2 and US#7 using this code.  But if it bothers you, you can close it, and we will re-open it again when/if we need it.

**Comment by jkarni** [comment]
Have we decided how integration is going to happen? My impression was that the meeting we had was inconclusive.

**Comment by ChristianSi** [comment]
I renamed it to "OBSOLETE" to make it clear that's it not supposed to be merged. OK?

**Comment by fisx** [comment]
this has been obsoleted by #358

**Pull request closed without merging**